index.js

```diff
@@ -0,0 +1,258 @@
+const userDb = require('./data/helpers/userDb.js');
+const postDb = require('./data/helpers/postDb.js');
+const express = require('express');
+const cors = require('cors');
+
+const server = express();
+server.use(cors());
+server.use(express.json());
+
+const toCaps = (request, response, next) => {
+    request.name = request.body.name.toUpperCase();
+    next();
+};
+
+const port = 8000;
+server.listen(port, () =>
+    console.log(`Server is listening to Port ${port}`)
+);
+
+// USER API
+server.get('/api/users', (request, response) => {
+    userDb
+        .get()
+        .then(users => {
+            return response
+                .status(200)
+                .json(users);
+        })
+        .catch(() => {
+            return response
+                .status(500)
+                .json({ Error: "Could not find list of users." })
+        });
+});
+
+server.get('/api/users/:id', (request, response) => {
+    const id = request.params.id;
+
+    userDb
+        .get(id)
+        .then(user => {
+            if (!user) {
+                return response
+                    .status(404)
+                    .json({ Error: "Could not find user." })
+            } else return response
+                .status(200)
+                .json(user);
+        })
+        .catch(() => {
+            return response
+                .status(500)
+                .json({ Error: "User info could not be retrieved." })
+        });
+});
+
+server.post('/api/users', toCaps, (request, response) => {
+    const name = request.name;
+    const newUser = { name };
+
+    if (!newUser.name) {
+        return response
+            .status(400)
+            .send({ Error: "Please enter a name for the user" });
+    } else if (newUser.name.length > 128) {
+        return response
+            .status(400)
+            .send({ Error: "User name must be 128 or less characters" });
+    }
+
+    userDb
+        .insert(newUser)
+        .then(userID => {
+            const { id } = userID;
+            userDb
+                .get(id)
+                .then(user => {
+                    return response
+                        .status(201)
+                        .json(user);
+                });
```

Comment:
How could you handle the response if for some reason the db cannot retrieve the new user (but the insertion was successful)?